### ISSUE TITLE
[refactor]: improve i18n keys

### ISSUE THREAD

**Issue by BlackHole1** [netless-io/flat#664]
I feel that the names of many keys are not unique enough, do we need to introduce the concept of namespace. @Leooeloel @crimx 

e.g:

```
i-know => i-know-extra-countdown
```

```
(namespace name)room-exit:
    i-know
```

_Originally posted by @BlackHole1 in https://github.com/netless-io/flat/issues/662#issuecomment-848435607_

**Comment by joshua7v** [comment]
while working with an i18n project
language tags will easily turn into a mess, people will keep add on more unique tags to avoid collision

namespaces are good if you have a large number of tags. 
separate a 5000 lines file into smaller pieces will make life easier.
the scale of this project is not that big, yet. (which is under 400 lines for now).

I tend to make translation tags always map to its English version, and add some kind of modifiers if needed.

for example:

```
// copied from that PR
...
  "send": "send",  // <--------- 1
  "teacher": "(Teacher)", // <------ 2
  "unban": "Unmuted", // <------ 3
  "users": "user list", // <------ 3
  "cancel": "Cancel", // <--------- 1
  "client-to-edge-server-network-latency": "Network latency from client to edge server", // <--------- 4
  "close-option": "Close option",
  "confirmation-of-the-end-of-classes": "Make sure to end the class", // <--------- 4
...

1. this is what I will typically do, I will capitalize it though. 
I tend to capitalize the first word of a sentence because you can easily transform it to uppercase/lowercase with CSS
but not as easy to capitalize the first word only.
sentence example: 'Network latency from client to edge server'

2. teacher -> 老师 seems could be reused somewhere else(which might not need parentheses,
so I might take off the parentheses in language tags, and do it some other way

3. unban -> Unmuted, users -> user list, English translations and translation tags does not match,
I'd like to make them match, unmuted -> Unmuted, user_list -> User list,
'users' is a pretty common one, you would need 'users -> 用户' instead of 用户列表,

4. I don't treat language tags as 'variables', they are the sentence representations, 
again to make them match should be more readable and consistent
t('client-to-edge-server-network-latency') -> t('network_latency_from_client_to_edge_server')

```


above are my personal thoughts on i18n translations.
willing to hear a good/better solution.

**Comment by BlackHole1** [comment]
I personally tend to use namespace because it has the following advantages:

1. Easier maintenance at the end. Instead of waiting for the number of rows to increase to a certain value before doing.
2. Semantic Differences. eg:
    `cancel`:
        in room page: `取消`
        in create page: `撤销`
    Through namespace, the semantic differences in different scenarios can be effectively resolved


I agree with your comments 2, 3 and 4.

One of the first point you said, I for personal, I think this is not so good, This is because it would further couple i18n to the business.
Personally, I believe that i18n should only be used to translate text and should not be responsible for considering other events.
If this were done, then all the places where i18n is used would go to maintain a new css style to solve the case problem.

**Comment by BlackHole1** [comment]
@crimx @hyrious @Cheerego7 What are your thoughts?

**Comment by crimx** [comment]
Namespace was considered initially but we found it tricky to educate non-tech members (who were heavily involved in the early stage of flat-i18n) to use [i18n Ally](https://github.com/lokalise/i18n-ally) namespace.

So we dropped namespace for simplicity reason. Although it's not a nice solution, it works fine at least for now. Namespace is definitely something we want to put more thoughts into once we finish the remaining major features.
